WangscapeGui/mainwindow.cpp

```diff
@@ -0,0 +1,66 @@
+#include "mainwindow.h"
+#include "ui_mainwindow.h"
+
+#include <iostream>
+
+#include <iostream>
+#include <fstream>
+#include <memory>
+
+#include <boost/program_options/options_description.hpp>
+#include <boost/program_options/parsers.hpp>
+#include <boost/program_options/variables_map.hpp>
+
+#include "logging/Logger.h"
+#include "logging/Logging.h"
+#include "logging/ConsoleAppender.h"
+#include "logging/FileAppender.h"
+#include "tilegen/TilesetGenerator.h"
+#include "tilegen/TileGenerator.h"
+#include "tilegen/partition/TilePartitionerNoise.h"
+
+#include "OptionsManager.h"
+
+namespace po = boost::program_options;
+
+std::string usage(std::string program_path)
+{
+    return "Usage: " + program_path + " rel/path/to/options.json\n";
+}
+
+MainWindow::MainWindow(QWidget *parent) :
+    QMainWindow(parent),
+    ui(new Ui::MainWindow)
+{
+    ui->setupUi(this);
+    connect(ui->pushButton, SIGNAL(pressed()), this, SLOT(clickPushButton()));
+}
+
+void MainWindow::clickPushButton()
+{
+    logging::addAppender(std::make_unique<logging::ConsoleAppender>("console", logging::Level::Debug));
```

Comment:
I think while it's in the experimental stage we should use a logger. Log output should be viewable through something like a log widget or nonmodal window before public release.